pyvista/core/pointset.py

```diff
@@ -339,7 +341,167 @@ def rotate_vector(self, *args, **kwargs):
+        ``True``. For performance, set this to ``False``.
+
+    deep : bool, optional
+        Whether to copy the input ``points``, or to create a PointSet from them
+        without copying them.  Setting ``deep=True`` ensures that the original
+        arrays can be modified outside the mesh without affecting the
+        mesh. Default is ``False``.
+
+    force_float : bool, optional
+        Casts the datatype to ``float32`` if points datatype is non-float.
+        Default ``True``. Set this to ``False`` to allow non-float types,
+        though this may lead to truncation of intermediate floats when
+        transforming datasets.
+
+    Notes
+    -----
+    This class requires ``vtk>=9.1.0``. This is an abstract class in
+    ``vtk<9.1.0`` and cannot be instantiated.
+
+    Examples
+    --------
+    Create a simple point cloud of 10 points from a numpy array.
+
+    >>> import numpy as np
+    >>> import pyvista
+    >>> rng = np.random.default_rng()
+    >>> points = rng.random((10, 3))
+    >>> pset = pyvista.PointSet(points)
+
+    Plot the pointset. Note: this casts to a :class:`pyvista.PolyData`
+    internally when plotting.
+
+    >>> pset.plot(point_size=10)
+
+    """
+
+    def __init__(self, points=None, editable=True, deep=False, force_float=True):
+        """Initialize the pointset."""
+        if pyvista.vtk_version_info < (9, 1, 0):
+            raise VTKVersionError("pyvista.PointSet requires VTK >= 9.1.0")
+
+        super().__init__()
+        if points is not None:
+            self.SetPoints(pyvista.vtk_points(points, deep=deep, force_float=force_float))
+
+        self.editable = editable
+
+    def __repr__(self):
+        """Return the standard representation."""
+        return DataSet.__repr__(self)
+
+    def __str__(self):
+        """Return the standard str representation."""
+        return DataSet.__str__(self)
+
+    @property
+    def points(self) -> pyvista_ndarray:
+        """Return a reference to the points as a numpy object.
+
+        Notes
+        -----
+        Points may only be set when the ``editable`` attribute is ``False``.
+
+        Examples
+        --------
+        Create an empty ``pyvista.PointSet`` and assign points to it.
+
+        >>> import numpy as np
+        >>> import pyvista
+        >>> pset = pyvista.PointSet()
+        >>> rng = np.random.default_rng()
+        >>> pset.points = rng.random((10, 3))
+
+        """
+        points = super().points
+        points.flags.writeable = self.editable
+        return points
+
+    @points.setter
+    def points(self, points: Union[VectorArray, NumericArray, pyvista._vtk.vtkPoints]):
+        if not self.editable:
+            raise ValueError("PointSet is read only as editable is set to False")
+
+        # setter inheritance workaround:
+        # https://stackoverflow.com/questions/10810369
+        super(self.__class__, self.__class__).points.fset(self, points)
```

Comment:
Slight nitpick: we could avoid using dunders here. And a more general note: I think this would break if we were to subclass `PointSet`. But we don't plan to, so this is probably fine. If I understand correctly the expected failure mode would be infinite recursion when accessing `self.points`, which is not something that is easily missed.
```suggestion
        super(type(self), type(self)).points.fset(self, points)
```